boto/resultset.py

```diff
@@ -19,10 +19,13 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
 # IN THE SOFTWARE.
 
+from collections import Sequence
+from boto.vendored.six.moves import UserList
+
 from boto.s3.user import User
 
 
-class ResultSet(list):
+class ResultSet(UserList):
```

Comment:
Yeah, that's a valid concern on the API. I'll see what i can do from Jython side, but I'd like to keep it open for a bit -  maybe someone else will chime in.